watertap/property_models/NaCl_prop_pack.py

```diff
@@ -741,49 +741,34 @@ def calculate_scaling_factors(self):
             iscale.set_scaling_factor(self.enth_flow,
                                       iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'H2O'])
                                       * iscale.get_scaling_factor(self.enth_mass_phase['Liq']))
-
         # transforming constraints
-        # property relationships with no index, simple constraint
-        v_str_lst_simple = ('osm_coeff', 'pressure_osm')
-        for v_str in v_str_lst_simple:
-            if self.is_property_constructed(v_str):
-                v = getattr(self, v_str)
-                sf = iscale.get_scaling_factor(v, default=1, warning=True)
-                c = getattr(self, 'eq_' + v_str)
-                iscale.constraint_scaling_transform(c, sf)
+        if self.is_property_constructed('pressure_osm'):
+            sf = iscale.get_scaling_factor(self.pressure_osm, default=1, warning=True)
+            iscale.constraint_scaling_transform(self.eq_pressure_osm, sf)
+        if self.is_property_constructed('osm_coeff'):
+            sf = iscale.get_scaling_factor(self.mass_frac_phase_comp['Liq', 'NaCl'], default=1, warning=True)
+            iscale.constraint_scaling_transform(self.eq_osm_coeff, sf**1.25)
 
         # property relationships with phase index, but simple constraint
-        for v_str in ('flow_vol_phase', 'visc_d_phase', 'diffus_phase'):
+        for v_str in ('visc_d_phase', 'enth_mass_phase', 'flow_vol_phase', 'diffus_phase'):
             if self.is_property_constructed(v_str):
-                v = getattr(self, v_str)
-                sf = iscale.get_scaling_factor(v['Liq'], default=1, warning=True)
-                c = getattr(self, 'eq_' + v_str)
-                iscale.constraint_scaling_transform(c, sf)
+                sf = iscale.get_scaling_factor(self.component(v_str)['Liq'], default=1, warning=True)
+                iscale.constraint_scaling_transform(self.component('eq_'+v_str), sf)
 
-        for v_str in ('dens_mass_phase', 'enth_mass_phase'):
-            if self.is_property_constructed(v_str):
-                v = getattr(self, v_str)
-                sf = iscale.get_scaling_factor(v['Liq'], default=1, warning=True)
-                c = getattr(self, 'eq_' + v_str)
-                iscale.constraint_scaling_transform(c, sf*10.)
+        if self.is_property_constructed('dens_mass_phase'):
+            sf = iscale.get_scaling_factor(self.mass_frac_phase_comp['Liq', 'NaCl'])
+            iscale.constraint_scaling_transform(self.eq_dens_mass_phase, sf/self.params.dens_mass_param['1'])
 
         # property relationship indexed by component
-        v_str_lst_comp = ['molality_comp']
-        for v_str in v_str_lst_comp:
```

Comment:
I created these blocks to transform similar constraints to make it easier on the developer. I know in this case there is only one property that is indexed by component, but possibly in the future a developer might add another and it would be easier for them to just add to this list rather than create the 4 lines of code. It's not a big deal though and we can get rid of it, but that was my intention.